internal/git/catfile.go

```diff
@@ -69,9 +76,14 @@ func (r *Repo) GetRefs(opts *GetRefs) ([]*GetRefsItem, error) {
 			return nil, errors.Wrap(err, "failed to read cat-file output")
 		}
 		item.Contents = make([]byte, size)
-		_, err = io.ReadFull(output, item.Contents)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to read cat-file output")
+		if n, err := io.ReadFull(output, item.Contents); err != nil {
+			logrus.WithFields(logrus.Fields{
```

Comment:
Ended up needing to debug this (tl;dr: `GitSdin` was deleting the final newline which caused reading exactly `size` bytes to fail -- that's why I just deleted `GitStdin`, only this function was using it).